### DISCUSSION TITLE
Cookies in DgsSubscription

### DISCUSSION THREAD

**Discussion by shikhar-acko** [Netflix/dgs-framework#1142] [Q&A]
Hi, 
There is a requirement to access cookies in DGS Subscription. I have tried accessing it using DgsDataFetcher but the request data inside DgsContext doesn't have the passed cookies.

Please help on how I can access 'cookie' in DgsSubscription

Thanks
Shikhar

**Comment by srinivasankavitha** [comment]
You should be able to use the annotation in your data fetcher  as described here: https://netflix.github.io/dgs/datafetching/#requestheader-requestparam-and-cookievalue

**Reply by rishabh-jha708** [reply]
Hi @srinivasankavitha , when using subscriptions I was not able to get headers, Annotation was throwing NullPointerException

**Reply by srinivasankavitha** [reply]
Sorry, missed the part about subscriptions. The annotations are not supported for subscriptions yet unfortunately. We will need to add this functionality.

**Reply by juferdinand** [reply]
Exists any updates?

**Reply by srinivasankavitha** [reply]
@juferdinand - we haven't added this feature yet. We do accept contributions if anyone is interested in adding it. I don't think we will be able to prioritize this request anytime soon since we don't have a use case internally yet.
